frontend/pages/hosts/ManageHostsPage/components/HostsDataTable/HostsDataTable.jsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useTable } from 'react-table';
+
+import hostInterface from 'interfaces/host';
+import { humanHostMemory, humanHostUptime } from 'kolide/helpers';
+import TextCell from '../TextCell/TextCell';
+import StatusCell from '../StatusCell/StatusCell';
+import LinkCell from '../LinkCell/LinkCell';
+
+// This data table uses react-table for implementation. The relevant documentation of the library
+// can be found here https://react-table.tanstack.com/docs/api/useTable
+const HostsDataTable = (props) => {
+  const { hosts } = props;
+  const columns = React.useMemo(() => {
+    return [
+      { Header: 'Hostname', accessor: 'hostname', Cell: cellProps => <LinkCell value={cellProps.cell.value} host={cellProps.row.original} /> },
+      { Header: 'Status', accessor: 'status', Cell: cellProps => <StatusCell value={cellProps.cell.value} /> },
+      { Header: 'OS', accessor: 'os_version', Cell: cellProps => <TextCell value={cellProps.cell.value} /> },
+      { Header: 'Osquery', accessor: 'osquery_version', Cell: cellProps => <TextCell value={cellProps.cell.value} /> },
+      { Header: 'IPv4', accessor: 'primary_ip', Cell: cellProps => <TextCell value={cellProps.cell.value} /> },
+      { Header: 'Physical Address', accessor: 'primary_mac', Cell: cellProps => <TextCell value={cellProps.cell.value} /> },
+      { Header: 'CPU', accessor: 'host_cpu', Cell: cellProps => <TextCell value={cellProps.cell.value} /> },
+      { Header: 'Memory', accessor: 'memory', Cell: cellProps => <TextCell value={cellProps.cell.value} formatter={humanHostMemory} /> },
+      { Header: 'Uptime', accessor: 'uptime', Cell: cellProps => <TextCell value={cellProps.cell.value} formatter={humanHostUptime} /> },
+    ];
+  }, []);
```

Comment:
Not sure if this is a react-table pattern or something you came up with, but I like it!

Reply:
ha thanks. its a bit of both. been using this pattern for a while, but react-table has it in their docs too. Its a good one when you have possibility of a lot of dynamic renders (in this case, many different type of table cell UIs)